apps/address-book/app/components/App/App.js

```diff
@@ -1,17 +1,16 @@
 import { observe, SidePanel, Main, AppBar, AppView } from '@aragon/ui'
 import PropTypes from 'prop-types'
 import React from 'react'
-import { hot } from 'react-hot-loader'
 import styled from 'styled-components'
-import { map } from 'rxjs/operators'
 import Entities from './Entities'
 import NewEntity from '../Panel/NewEntity'
 import { networkContextType, AppTitle, AppTitleButton } from '../../../../../shared/ui'
+import { useAragonApi } from '@aragon/api-react'
+import { IdentityProvider } from '../../../../../shared/identity'
 
 class App extends React.Component {
   static propTypes = {
-    app: PropTypes.object.isRequired,
-    // TODO: Shape this
+    api: PropTypes.object,
```

Comment:
This is required, right? Won't things break if it's not present?

Reply:
Actually no - initially there is no api, it is provided slightly later. If "isRequired" is set, there is a big warning in the console.

Reply:
But the code in the component assumes that it's there, so it seems wrong to make it unrequired.

Maybe we could set `defaultProps` with an empty object, and add a comment there that `useAragonApi` doesn't provide `api` on initial load. Then we'll also have a reminder in the future to remove our `defaultProps` if `useAragonApi` fixes this inconsistency.

Reply:
Actually, it can't be provided with defaultProps, because it is provided (albeit, initially, null):
```
Warning: Failed prop type: The prop `api` is marked as required in `App`, but its value is `null`.
    in App (created by AppWrap)
    in AppWrap
    in AragonApi
```
This is after adding defaultProps and adding isRequired. So, what is being asserted now (I mean before I readded 'isRequired') is that if api is present and not null it should be an object. Soft requirement... We could in theory to this:

`PropTypes.oneOfType([null, PropTypes.object]).isRequired`

But it doesn't strike me as perfect solution. Is it the least wrong solution?

Reply:
I don't think that provides more information than leaving it optional. Thanks for looking into this. I'd like to see `@aragon/api-react` change this behavior in the future, but I think the way you implemented it initially is the best for now.